src/api/firebase.js

```diff
@@ -235,3 +235,56 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(list) {
+	const currentDate = new Date();
+	const soon = [];
+	const kindOfSoon = [];
+	const notSoon = [];
+	const inactive = [];
+	const overdue = [];
+
+	const sortedList = list.sort((a, b) => {
+		const dateNextPurchasedAsDateA = a.dateNextPurchased?.toDate();
+		const dateNextPurchasedAsDateB = b.dateNextPurchased?.toDate();
+
+		const daysUntilNextPurchaseA = getDaysBetweenDates(
+			currentDate,
+			dateNextPurchasedAsDateA,
+		);
+		const daysUntilNextPurchaseB = getDaysBetweenDates(
+			currentDate,
+			dateNextPurchasedAsDateB,
+		);
+
+		return daysUntilNextPurchaseB > daysUntilNextPurchaseA ? -1 : 1;
```

Comment:
nitpick: I’m wondering if it’s necessary to use `getDaysBetweenDates` here. If the purpose of the `sortedList `function is to return a list sorted by each item's `dateNextPurchased`, how about just sorting based on this property  like
```
 const sortedList = list.sort((a, b) => {
    const dateNextPurchasedA = a.dateNextPurchased?.toDate();
    const dateNextPurchasedB = b.dateNextPurchased?.toDate();

    return dateNextPurchasedA - dateNextPurchasedB;
});
```

Reply:
Ooo, really cool idea Emma! That makes a lot of sense and I like that it's simple. 

Also, I tried it out and it seems a bit buggy with organization. Since the idea is that we need it to be organized by how far out `dateNextPurchased` is from the current day, using `getDaysBetweenDates` makes that very clear.